### PR TITLE
operator callback and workspaces cleanup and fixes

### PR DESCRIPTION
**Pull request by imanjra**
## What changes are proposed in this pull request?

- Load workspace on save
- Add loading state to delete and save
- code cleanup
- Style tweaks

## How is this patch tested? If it is not, please explain why.

Using the workspaces in the app

### Is this a user-facing change that should be mentioned in the release notes?

-   [x] No. You can skip the rest of this section.
-   [ ] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

(Details in 1-2 sentences. You can just refer to another PR with a description
if this PR is part of a larger change.)

### What areas of FiftyOne does this PR affect?

-   [ ] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Summary by CodeRabbit

- **New Features**
	- Enhanced operator execution by introducing a new callback functionality for improved interactivity and responsiveness.
	- Added new constants for workspace operations (load, save, delete, list) in the `Workspaces` component.
	- Renamed components from `SavedSpaces` to `Workspaces` and `SavedSpace` to `Workspace` for clarity and consistency.
	- Introduced a new `MuiButton` component with a loading indicator for enhanced user experience.

- **Bug Fixes**
	- Ensured functionality consistency by updating function calls and hooks to align with new component and operator names.

- **Refactor**
	- Utilized the newly introduced `MuiButton` component in modified components and functions.
	- Replaced deprecated hooks and methods with updated ones in the workspace management components.

- **Documentation**
	- Updated component names and descriptions to reflect the renaming and new features in workspace management tools.

### CODE REVIEW

**Comment by brimoor** [comment]
@imanjra I'll defer to others for code review but I just tested this PR and everything looks and works great!

With respect to the positioning of the `. Unsaved [icon]` component, I definitely prefer the new layout compared to the old one. Maybe decrease the right padding just a bit more so the right edge of the workspace icon aligns with the right edge of the grid reset icon?

Screenshot 2024-05-07 at 6 17 08 PM

File: app/packages/operators/src/operators.ts
Comment (ritch) [reply app/packages/operators/src/operators.ts]: Isn't the callback optional?

File: app/packages/operators/src/state.ts
Comment (ritch) [inline app/packages/operators/src/state.ts:889]: Why is `callback` being added in this PR?

File: app/packages/spaces/src/components/Workspaces/WorkspaceEditor.tsx
Comment (ritch) [inline app/packages/spaces/src/components/Workspaces/WorkspaceEditor.tsx:113]: Is it so that you can use it here?

**Review by ritch — APPROVED** [review]
Implementation LGTM. Please rebase py-panels-develop after merging this because it will likely have conflicts.

File: app/packages/spaces/src/components/Workspaces/WorkspaceEditor.tsx
Comment (imanjra) [reply app/packages/spaces/src/components/Workspaces/WorkspaceEditor.tsx:113]: Yah, I refactored the workspace code little bit to use `executeOperator` with the new callback instead of `useOperatorExecutor`

**Pull request merged**
